src/modm/platform/core/cortex/module.lb

```diff
@@ -315,6 +315,13 @@ def validate(env):
     flash_reserved_option_name = "modm:platform:cortex-m:linkerscript.flash_reserved"
     flash_reserved = env[flash_reserved_option_name]
     if flash_reserved != 0:
+        if flash_reserved % 8192 != 0:
+            raise ValidateException("Invalid reserved flash size in option '{}'. "
+                                    "The 'linkerscript.flash_reserved' of {} "
+                                    "is not a multiple of 8Kbytes, which could "
+                                    " lead to accidential program flash erasure."
+                                    .format(flash_reserved_option_name,
+                                            flash_reserved))
```

Comment:
STM32 has different sizes for flash pages, so this check is not scalable. There might also be use-cases that would warrant non-aligned reservation sizes (like merging two binaries into one).

Reply:
@rleh had a strong opinion about this one so that's why I added it. 
I myself think that it should be user responsibility to only reserve multiples of the given flash page size and/or take other precautions to not erase program data otherwise, so feel free to drop the last commit.

Regarding STM32's different flash sizes: the 8Kbytes should be an integer multiple of all possible STM32 flash page sizes, thus work for all models (if you want to be extra sure, maybe 16Kbytes are a safer option).

As of intentionally wanting non-aligned reserved sizes, @rleh was of the opinion that this would be rare and/or user responsibility to hack that together.

Reply:
I think this should be the responsibility of a flash storage module that then probably also has other restrictions like a minimum number of flash pages. For example, [littlefs](https://github.com/littlefs-project/littlefs/blob/master/DESIGN.md) requires at least a few flash pages to work correctly, so this requirement is already too weak. Let's add this later once we have a concrete use case. 

I also recently discovered there is a database inside the [STM32CubeProgrammer](https://www.st.com/en/development-tools/stm32cubeprog.html) with all the memory sizes and flash pages, so we could generate the flash size code in the `:platform:flash` module and use it for queries like this.